### ISSUE TITLE
`assert_never()` throws error on Python 3.10

### ISSUE THREAD

**Issue by johnthagen** [python/mypy#12613]
**Bug Report**

`assert_never()` does not seem to work correctly on Python 3.10 with the latest `mypy` and `typing-extensions`.

**To Reproduce**

```py
from typing_extensions import assert_never


def int_or_str(arg: int | str) -> None:
    match arg:
        case int():
            print("It's an int")
        case str():
            print("It's a str")
        case _ as unreachable:
            assert_never(unreachable)
```

But get an unexpected error:

```
$ mypy main.py
main.py:1: error: Module "typing_extensions" has no attribute "assert_never"
Found 1 error in 1 file (checked 1 source file)
```

The definition I see within `typing-extensions` has a conditional, but is none-the-less there:

```py
if hasattr(typing, "assert_never"):
    assert_never = typing.assert_never
else:
    def assert_never(__arg: Never) -> Never:
        ...
```

**Expected Behavior**

No error and for type checking to be successful.

**Your Environment**

- Python 3.10.3
- Windows 10

```
mypy              0.942
mypy-extensions   0.4.3
typing_extensions 4.2.0
```

```toml
[tool.mypy]
ignore_missing_imports = true
strict = true
```

**Comment by JelleZijlstra** [comment]
Mypy doesn't look at the runtime code in typing-extensions, but at the stubs in https://github.com/python/typeshed/blob/master/stdlib/typing_extensions.pyi. The stubs update adding assert_never() must not have made it into the last mypy release yet. It will be in the next one.

For assert_never specifically, you can just define it in your own code directly with the same signature as in typing-extensions, since there's no magic around it.

**Comment by johnthagen** [comment]
I can confirm this is fixed in Mypy 0.950.
